docs/design/architecture.md

```diff
@@ -2,128 +2,353 @@
+The Kata Containers runtime (or shim), is a [shimv2](#shim-v2-architecture) compatible runtime.
 
-`containerd-shim-kata-v2` heavily utilizes the
-[virtcontainers package](../../src/runtime/virtcontainers/), which provides a generic, runtime-specification agnostic, hardware-virtualized containers library.
+The name of the Kata Containers shim binary is
+[`containerd-shim-kata-v2`](../../src/runtime/cmd/containerd-shim-kata-v2).
 
-### Configuration
+The runtime makes heavy use of the [`virtcontainers`
+package](../../src/runtime/virtcontainers), which provides a generic,
+runtime-specification agnostic, hardware-virtualized containers
+library.
+
+The runtime is responsible for starting the [hypervisor](#hypervisor)
+and it's VM, and communicating with the [agent](#agent) using a
+[ttRPC based protocol](#agent-communications-protocol) over a VSOCK
+socket that provides a communications link between the VM and the
+host. 
+
+This protocol allows the runtime to send container management commands
+to the agent. The protocol is also used to carry the standard I/O
+streams (`stdout`, `stderr`, `stdin`) between the containers and
+container managers (such as CRI-O or containerd).
 
-The runtime uses a TOML format configuration file called `configuration.toml`. By default this file is installed in the `/usr/share/defaults/kata-containers` directory and contains various settings such as the paths to the hypervisor, the guest kernel and the mini-OS image.
+## Utility program
 
-The actual configuration file paths can be determined by running:
-```
-$ kata-runtime --show-default-config-paths
-```
-Most users will not need to modify the configuration file.
+The `kata-runtime` binary is a utility program that provides
+administrative commands to manipulate and query a Kata Containers
+installation.
 
-The file is well commented and provides a few "knobs" that can be used to modify the behavior of the runtime and your chosen hypervisor.
+> **Note:**
+>
+> In Kata 1.x, this program also acted as the main
+> [runtime](#runtime), but this is no longer required due to the
+> improved shimv2 architecture.
+
+### Configuration
+
+See the [configuration file details](../../src/runtime/README.md#configuration).
 
 The configuration file is also used to enable runtime [debug output](../Developer-Guide.md#enable-full-debug).
 
+## Process overview
+
+The table below shows an example of the main processes running when a
+Kata Container is created with containerd:
+
+| Description | Host | VM root context | VM container context |
```

Comment:
I mentioned this above but I think someone would be confused by the VM root context and VM container context. I think adding in what the user has access to and what they don't would help as well.

Reply:
Added lots more detail about this in the "Environments" section.